R/expr__expr.R

```diff
@@ -2441,10 +2407,13 @@ Expr_limit = function(n = 10) {
 
 
 
-#' Exponentiation `^` or `**`
+#' Exponentiation `^` and `**`.
 #' @description Raise expression to the power of exponent.
 #' @keywords Expr
 #' @param exponent exponent
+#' @details in py-polars python `^` is the xor operator. The R interpreter will replace the
+#' `**` with `^`, such that `**` means `^`. Except e.g. in strings like "**". Use of `**` will
+#' not work in some cases, because it is not a real separate operator in R.
```

Comment:
```suggestion
#' @details In python-polars, `^` is the xor operator. The R interpreter will replace the
#' `^` with `**`, which means that you can use both `**` and `^` for the exponentiation. Use of `**` will
#' not work in some cases, because it is not a real separate operator in R.
```
Are my corrections what you meant?

Also, you say "Use of `**` will not work in some cases, because it is not a real separate operator in R." 
-> Not clear, when does it fail? Can we show examples of this?

Reply:
It is strange not well documented quirk of R 

I will add something like this to documentation.

``` r
#in a new session
2^1
#> [1] 2
2**1 #this works because ** is converted to the ^-operator  by R interpreter
#> [1] 2

get("^")(2,1) #this works because there exists a function called "^"
#> [1] 2
get("**")(2,1)  #the R interpreter will not convert "**" to "^"  and there is no function named "**"
#> Error in get("**"): object '**' not found
```

<sup>Created on 2023-08-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

looking up `?"**"`
** is translated in the parser to ^, but this was undocumented for many years. It appears as an index entry in Becker et al (1988), pointing to the help for Deprecated but is not actually mentioned on that page. Even though it had been deprecated in S for 20 years, it was still accepted in R in 2008.

Reply:
I don't think you should add the examples above in the docs, they are more related to base R than to `polars`. I think mentioning that `^` and `**` mean the same thing in `polars` expressions should be enough. If you have some use cases where using `**` doesn't work then they should be shown in examples.